beacon-chain/rpc/eth/events/events.go

```diff
@@ -556,113 +544,169 @@ func (s *Server) lazyReaderForEvent(ctx context.Context, event *feed.Event, topi
 	}
 }
 
-// This event stream is intended to be used by builders and relays.
-// Parent fields are based on state at N_{current_slot}, while the rest of fields are based on state of N_{current_slot + 1}
-func (s *Server) currentPayloadAttributes(ctx context.Context) (lazyReader, error) {
-	headRoot, err := s.HeadFetcher.HeadRoot(ctx)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not get head root")
+var errUnsupportedPayloadAttribute = errors.New("cannot compute payload attributes pre-Bellatrix")
+
+func (s *Server) computePayloadAttributes(ctx context.Context, ev payloadattribute.EventData) (payloadattribute.Attributer, error) {
+	v := ev.HeadState.Version()
+	if v < version.Bellatrix {
+		return nil, errors.Wrapf(errUnsupportedPayloadAttribute, "%s is not supported", version.String(v))
 	}
-	st, err := s.HeadFetcher.HeadState(ctx)
+
+	t, err := slots.ToTime(ev.HeadState.GenesisTime(), ev.HeadState.Slot())
 	if err != nil {
-		return nil, errors.Wrap(err, "could not get head state")
+		return nil, errors.Wrap(err, "could not get head state slot time")
 	}
-	// advance the head state
-	headState, err := transition.ProcessSlotsIfPossible(ctx, st, s.ChainInfoFetcher.CurrentSlot()+1)
+	timestamp := uint64(t.Unix())
+	prevRando, err := helpers.RandaoMix(ev.HeadState, chaintime.CurrentEpoch(ev.HeadState))
 	if err != nil {
-		return nil, errors.Wrap(err, "could not advance head state")
+		return nil, errors.Wrap(err, "could not get head state randao mix")
 	}
-
-	headBlock, err := s.HeadFetcher.HeadBlock(ctx)
+	proposerIndex, err := helpers.BeaconProposerIndex(ctx, ev.HeadState)
 	if err != nil {
-		return nil, errors.Wrap(err, "could not get head block")
+		return nil, errors.Wrap(err, "could not get head state proposer index")
 	}
-
-	headPayload, err := headBlock.Block().Body().Execution()
-	if err != nil {
-		return nil, errors.Wrap(err, "could not get execution payload")
+	feeRecpt := params.BeaconConfig().DefaultFeeRecipient.Bytes()
```

Comment:
i think this might be wrong if you set the suggested fee recipient flag for untracked validators, I think it needs to be the burn address

Reply:
This preserves the existing behavior: https://github.com/prysmaticlabs/prysm/blob/develop/beacon-chain/rpc/eth/events/events.go#L600-L604

If we want to change it I'd rather do that in a separate PR.